src/main/java/frc/trigon/robot/poseestimation/poseestimator/VisionObservationSorter.java

```diff
@@ -0,0 +1,74 @@
+package frc.trigon.robot.poseestimation.poseestimator;
+
+import java.util.Arrays;
+import java.util.function.ToDoubleFunction;
+
+public class VisionObservationSorter {
```

Comment:
If this works for any object:
1. Make this a utils class in the lib with an abstract name
2. Add a very good jdoc